File: src/main/java/play/pluv/feed/domain/Feed.java
package play.pluv.feed.domain;

import static jakarta.persistence.GenerationType.IDENTITY;
import static lombok.AccessLevel.PROTECTED;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import play.pluv.base.BaseEntity;

@ToString
@Entity
@NoArgsConstructor(access = PROTECTED)
@Getter
public class Feed extends BaseEntity {

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;
  private Long memberId;
  private String title;
  private String creatorName;
  private String artistNames;
  private String thumbNailUrl;
  private Boolean viewable;

  @Builder
  public Feed(final Long memberId, final String title,
      final String creatorName, final String artistNames,
      final String thumbNailUrl) {
    this.memberId = memberId;
    this.title = title;
    this.creatorName = creatorName;
    this.artistNames = artistNames;
    this.thumbNailUrl = thumbNailUrl;
    this.viewable = true;
    this.createdAt = LocalDateTime.now();
    this.updatedAt = LocalDateTime.now();
  }

  public Feed(
      final Long id, final Long memberId, final String title,
      final String creatorName, final String artistNames, final String thumbNailUrl
  ) {
    this.id = id;
    this.memberId = memberId;
    this.title = title;
    this.creatorName = creatorName;
    this.artistNames = artistNames;
    this.thumbNailUrl = thumbNailUrl;
    this.viewable = true;
    this.createdAt = LocalDateTime.now();
    this.updatedAt = LocalDateTime.now();
  }
}

File: src/main/kotlin/play/pluv/feed/domain/Feed.kt
package play.pluv.feed.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import play.pluv.base.BaseEntity
import java.time.LocalDateTime

@Entity
class Feed(
    memberId: Long,
    title: String,
    creatorName: String,
    artistNames: String,
    thumbNailUrl: String,
    songCount: Int,
) : BaseEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @get:JvmName("getNullableId")
    var id: Long? = null
    var memberId: Long = memberId
        private set
    var title: String = title
        private set
    var creatorName: String = creatorName
        private set
    var artistNames: String = artistNames
        private set
    var thumbNailUrl: String = thumbNailUrl
        private set
    var viewable: Boolean = true
        private set
    var songCount: Int = songCount
        private set

    fun getId(): Long = requireNotNull(id)

    constructor(
        id: Long,
        memberId: Long,
        title: String,
        creatorName: String,
        artistNames: String,
        thumbNailUrl: String,
        songCount: Int
    ) : this(memberId, title, creatorName, artistNames, thumbNailUrl, songCount) {
        this.id = id
        this.createdAt = LocalDateTime.now()
        this.updatedAt = LocalDateTime.now()
    }
}
